src/views/DPI/components/DpiPriceChanges.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ReactNode } from 'react'
+import styled from 'styled-components'
+import numeral from 'numeral'
+
+import InfoSection from './InfoSection'
+
+import useDpiTokenMarketData from 'hooks/useDpiTokenMarketData'
+
+const DpiPriceChanges: React.FC = () => {
+  const { prices } = useDpiTokenMarketData()
+  const [startTime] = prices?.[0] || [0]
+  const [latestTime, latestPrice] = prices?.[prices.length - 1] || [0, 0]
+  // Coingecko API returns each item as hourly data if < 90 days and daily data > 90 days
+  // So if time between start and end time < 90 days, 1 day of data is 24 array items.
+  const dataCadenceFor24Hours = latestTime - startTime < 7776000000 ? 24 : 1
+  const priceChangeIntervals = [
+    ['1 Day', 1],
+    ['1 Month', 30],
+    ['3 Months', 90],
+    ['1 Year', 365],
+  ]
+
+  const calculatePriceDiffs = (daysOfComparison: number) => {
+    // placeholder if no data available for timerange
+    if (!prices || prices.length < daysOfComparison * dataCadenceFor24Hours)
+      return <div>---</div>
```

Comment:
lets wrap these in braces unless it's strictly a 1 liner (e.g. fits on line 25 here) and is below 120 characters